frontend/public/components/modals/cluster-update-modal.tsx

```diff
@@ -18,107 +19,77 @@ import {
   ModalTitle,
 } from '../factory/modal';
 
-class ClusterUpdateModal extends PromiseComponent<
-  ClusterUpdateModalProps,
-  ClusterUpdateModalState
-> {
-  readonly state: ClusterUpdateModalState;
-
-  constructor(public props: ClusterUpdateModalProps) {
-    super(props);
-    const available = getSortedUpdates(props.cv);
-    this.state.selectedVersion = _.get(available, '[0].version', '');
-  }
-
-  _submit = (e: React.FormEvent<EventTarget>) => {
+const ClusterUpdateModal = withHandlePromise((props: ClusterUpdateModalProps) => {
+  const { cancel, close, cv, errorMessage, handlePromise, inProgress } = props;
+  const [desiredVersion, setDesiredVersion] = React.useState(
+    getSortedUpdates(cv)[0]?.version || '',
+  );
+  const [error, setError] = React.useState(errorMessage);
+  const currentVersion = getDesiredClusterVersion(cv);
+  const availableSortedUpdates = getSortedUpdates(cv);
+  const dropdownItems = _.reduce(
+    availableSortedUpdates,
+    (acc, { version }) => {
+      acc[version] = version;
+      return acc;
+    },
+    {},
+  );
+  const submit: React.FormEventHandler<HTMLFormElement> = (e) => {
     e.preventDefault();
-    const { selectedVersion } = this.state;
-    if (!selectedVersion) {
+    if (!desiredVersion) {
       return;
     }
-    const { cv } = this.props;
     const available = getAvailableClusterUpdates(cv);
-    const desired = _.find(available, { version: selectedVersion });
+    const desired = _.find(available, { version: desiredVersion });
     if (!desired) {
-      this.setState({
-        errorMessage: `Version ${selectedVersion} not found among the available updates. Select another version.`,
-      });
+      setError(
+        `Version ${desiredVersion} not found among the available updates. Select another version.`,
```

Comment:
Needs to be externalized?

Reply:
I didn't externalize this because the other error messages are coming from the API and will be in english.  My assumption is its better to keep these all the same?